### PR TITLE
jujutsu: Remove superfluous shell completion

### PR DESCRIPTION
**Pull request by jcf**
This issue has come up before, and the recommendation was to remove the shell completion logic from Home Manager. This pull request now does just that.

- https://github.com/martinvonz/jj/pull/2945
- https://github.com/nix-community/home-manager/pull/5016#issuecomment-1947449541
- https://github.com/nix-community/home-manager/pull/5037#pullrequestreview-1888843990

### Description

``` sh
$ for sh (--bash --fish --zsh); jj util completion "$sh" >/dev/null
Warning: `jj util completion --bash` will be removed in a future version, and this will be a hard error
Hint: Use `jj util completion bash` instead
Warning: `jj util completion --fish` will be removed in a future version, and this will be a hard error
Hint: Use `jj util completion fish` instead
Warning: `jj util completion --zsh` will be removed in a future version, and this will be a hard error
Hint: Use `jj util completion zsh` instead
```

### Checklist

- [ ] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

#### Maintainer CC

@shikanime

### CODE REVIEW

**Comment by jcf** [comment]
I've tested this locally by overriding the Jujutsu program like so:

``` nix
programs.jujutsu.enableZshIntegration = false;
programs.zsh.initExtra = ''
  source <(${pkgs.jujutsu}/bin/jj util completion zsh)
    compdef _jj ${pkgs.jujutsu}/bin/jj
'';
```

**Review by ambroisie — CHANGES_REQUESTED** [review]
Duplicate of #5016, and same issue: this shouldn't even be in Home-Manager.

**Comment by jcf** [comment]
It sounds like the resolution was to [remove this completion](https://github.com/nix-community/home-manager/pull/5016#issuecomment-1947449541) from Home-Manager. Would such a pull request be welcome?

**Comment by jcf** [comment]
I've pushed another commit to remove the shell completion entirely.

File: modules/programs/jujutsu.nix [hunk 1]
```diff
@@ -33,24 +33,6 @@ in {
...
-      type = types.bool;
-      default = true;
-      description = "Whether to enable Zsh integration.";
-    };
-
-    enableFishIntegration = mkOption {
-      type = types.bool;
-      default = true;
-      description = "Whether to enable Fish integration.";
-    };
```
Comment (ambroisie) [inline modules/programs/jujutsu.nix:53]: These should have a `mkRemoveModuleOption` to explain why they were removed.

Comment (ambroisie) [reply modules/programs/jujutsu.nix:53]: Also, please squash your commits.

Comment (rycee) [reply modules/programs/jujutsu.nix:53]: I've applied these suggestions. Thanks!

**Comment by rycee** [comment]
Thanks! Merged to master now 🙂

**Pull request merged**